Sources/Task/TaskRecover.swift

```diff
@@ -47,4 +47,18 @@ extension Task {
         return Task<SuccessValue>(future: future, cancellation: cancel)
         #endif
     }
+
+    public func recoverWithNewTask(upon executor: Executor, start startNextTask: @escaping(Error) -> Task<SuccessValue>) -> Task<SuccessValue> {
```

Comment:
The "recover" naming seems to assume that, unlike the first task, this one is _totally_ gonna pan out. I think that assumption was more warranted in the synchronous case that this is modeled after, but it's possibly misleading here. (And possibly so in the synchronous case too, since that is allowed to `throw` and so fail to recover.)

Some (less optimistic) alternative names:

- failingThat(upon:try:)
- fallback(upon:to:)
- or(upon:execute:)

These also improve readability in the event you have several fallbacks available. I dunno about you, but seeing a chain of recover with this, recover with that, no wait, recover with this other thing, would leave me a bit confused - if it legit recovers, you should only ever need one, right?